/** (C) Copyright 2018, Applied Physical Sciences Corp., A General Dynamics Company
 **
 ** Gravity is free software; you can redistribute it and/or modify
 ** it under the terms of the GNU Lesser General Public License as published by
 ** the Free Software Foundation; either version 3 of the License, or
 ** (at your option) any later version.
 **
 ** This program is distributed in the hope that it will be useful,
 ** but WITHOUT ANY WARRANTY; without even the implied warranty of
 ** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 ** GNU Lesser General Public License for more details.
 **
 ** You should have received a copy of the GNU Lesser General Public
 ** License along with this program;
 ** If not, see <http://www.gnu.org/licenses/>.
 **
 */

option optimize_for = SPEED;
option java_package = "com.aphysci.gravity.protobuf";

package gravity.protobuf;

message GravityMOOSConfig {
    optional string gravity_name        = 1;
    optional double heartbeat_frequency = 2 [default=0.5];
    
    optional string moos_server = 3 [default="localhost"];
    optional uint32 moos_port   = 4 [default=9000];
    
    repeated string gravity_publications = 5;
    repeated string moospbt_publications = 6;

    //repeated goby.moos.TranslatorEntry custom_mappings = 7;
}

